### ISSUE TITLE
Multiple geth crashes in the last 2 days.

### ISSUE THREAD

**Issue by Crypto2** [ethereum/go-ethereum#20672]
Hi there,

Over the past 2 days we've been getting geth node crashes right around the same time. They are all 1.9.9 (updated to 1.9.10 now so knock on wood might have fix it.) Is there a known issue or some kind of attack on the Ethereum network?

They are 4 nodes all on Linux boxes running a mix of Debian 9 and 10 64-bit and would crash within the same 30-60 minute timespan. Some have NVMe and some have SATA SSD storage.

#### System information

Geth version: 1.9.9-stable-01744997
OS & Version: Linux, Debian 9 and 10

Unfortunately they are run under cron so don't have backtraces for them.

**Comment by Crypto2** [comment]
Actually, just had one crash again; unfortunately it's only a partial backtrace since it was longer than my SSH screen buffer. I have it running again logging to a file so if it dumps again should get all of it.

[eth.txt](https://github.com/ethereum/go-ethereum/files/4206046/eth.txt)

**Comment by Crypto2** [comment]
Here is a new full dump, looks like it's reporting out of memory. The system this is on has 128G of RAM and geth is the only thing on it other than the core Linux system.

Command line is: geth --ipcdisable --rpc --rpcapi "eth,net,web3,personal,admin,debug" --cache 16384 --syncmode fast --gcmode archive

It has been restarted since and after about an hour has ballooned up to 95 GB of RAM usage (RES column in top.) I added some massive swap files as a potential band-aid but that's pretty dirty. No mention of oom in the kernel logs so it doesn't seem like the kernel killed it.

[eth2.txt](https://github.com/ethereum/go-ethereum/files/4206454/eth2.txt)

**Comment by holiman** [comment]
It seems to die while writing some gigantic response to an rpc-call. Any idea what the call was?

**Comment by Crypto2** [comment]
Possibly a debug trace? That's the only one I can think of offhand that might be huge. I'll see if I can narrow it down.

**Comment by Crypto2** [comment]
My best guess would be this one: https://etherscan.io/tx/0x911a1f703dab4942d8e88855a723651e171b632c5d15444e13c3ca93542f9c53

**Comment by holiman** [comment]
Do you mean that someone was doing a trace on that one? I was more wondering if you know more details about what rpc-call was made, to trigger such a massive response. Is it a public node that is accessible to the internet?

**Comment by Crypto2** [comment]
We were running a trace on it, we support ERC20 and contract deposits so we end up tracing most/all ether transactions to see if they are deposits to our users. They are private nodes only we use.

**Comment by fjl** [comment]
@holiman It sounds like this issue could be fixed by this idea for a tracer change you mentioned two weeks ago, where we wouldn't necessarily need to run the tracing code for every VM opcode to extract things like transfers or calls.

**Comment by Crypto2** [comment]
Yep, for now we just disabled contract ETH deposits so we don't have to trace. (Since ERC20's can be done without tracing just from the receipt.)

**Comment by gballet** [comment]
@Crypto2 could you share with us what kind of trace was running on that server, when you get the crash?

**Comment by Crypto2** [comment]
We were using the debug_traceTransaction JSON-RPC call  with these options:

$traceopts = array('disableMemory' => true, 'disableStorage' => true);
debug_traceTransaction('hash', $traceopts);

Since we stopped geth has been running 100%
